smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelLoader.java

```diff
@@ -669,60 +668,10 @@ private void parseMixins(LoadOperation.DefineShape operation) {
     private void parseCollection(ShapeId id, SourceLocation location, CollectionShape.Builder<?, ?> builder) {
```

Comment:
Can we merge collection / map / and union into a single method that handles them all? Looks like the same grammar and code now.

Reply:
Done